### ISSUE TITLE
Upgrading to 14.6.0 breaks column metadata. "Unexpected token `COLMETADATA`"

### ISSUE THREAD

**Issue by chriseaton** [tediousjs/tedious#1445]
**Software versions**
* Tedious: 14.6.0
* SQL Server: SQL Server 2016
* Node.js: 16.15.0

**Additional Libraries Used and Versions**
[chriseaton/rhino](https://github.com/chriseaton/rhino)

**Problem description**
As soon as we upgrade to 14.6.0, our tedious calls break with the following error: 
```
UnexpectedTokenError: Unexpected token `COLMETADATA` in `Login7TokenHandler`
    at Login7TokenHandler.onColMetadata (/app/node_modules/tedious/lib/token/handler.js:91:11)
    at Readable.<anonymous> (/app/node_modules/tedious/lib/token/token-stream-parser.js:26:33)
    at Readable.emit (node:events:527:28)
    at addChunk (node:internal/streams/readable:315:12)
    at readableAddChunk (node:internal/streams/readable:289:9)
    at Readable.push (node:internal/streams/readable:228:10)
    at next (node:internal/streams/from:98:31)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
Emitted 'error' event on Readable instance at:
    at emitErrorNT (node:internal/streams/destroy:157:8)
    at emitErrorCloseNT (node:internal/streams/destroy:122:3)
    at processTicksAndRejections (node:internal/process/task_queues:83:21)
```
When we force the version back to 14.5.0 the problem goes away. The release doesn't say anything about column metadata handling changing, so are unsure why this is happening.

Here is our relevant code that traps the events from a query - this works prior to 14.6.0.
https://github.com/chriseaton/rhino/blob/master/connected-query.js#L104-L179

**Expected behavior**
The query to execute and get column metadata OK.

**Actual behavior**
Crash before any events fire.

**Any other details that can be helpful**

**Comment by arthurschreiber** [comment]
We refactored some internals to use `async`/`await`, and this might somehow be related to that. The `COLMETADATA` token should never be part of the `LOGIN7` response, so I think we're switching the state somewhere too early/too late.

@mShan0 Mind taking a look?

**Comment by mShan0** [comment]
Hi @chriseaton, is it possible to provide any steps to reproduce this issue and the config options used? I'm struggling to reproduce this through both Rhino and Tedious.

**Comment by chriseaton** [comment]
I will try to come up with a quick script - stand by.

**Comment by chriseaton** [comment]
I can't reproduce it right away with `default` login... Where we were using it, we were using `ntlm`... I will need to setup something more advanced to get that going.

**Comment by chriseaton** [comment]
Alrighty - seems to be NTLM related. Here's the code I used. It works 100% on `default` auth, but switching to `ntlm` causes the error up above when the query is run (calls `execSql` under the hood).

We used the AdventureWorks (light) database bak from here: https://github.com/Microsoft/sql-server-samples/releases/download/adventureworks/AdventureWorks2016.bak

Under default it worked, when we granted a ntlm account - it stopped. Downgrading back to 14.5.0 ntlm worked again.

```js
const rhino = require('rhino');

let driver = new rhino({
    server: 'localhost',
    authentication: {
        type: 'default',
        options: {
            userName: 'sa',
            password: '************'
        }
    },
    options: {
        database: 'AdventureWorks',
        port: 1433,
        encrypt: true,
        trustServerCertificate: true,
        useColumnNames: true,
        enableArithAbort: true
    }
});

(async () => {
    let success = await driver.ping();
    console.log('success', success);
    await driver.query('SELECT * FROM Person.Person');  //boom! under ntlm
    await driver.destroy();
})();
```

**Comment by chriseaton** [comment]
Also can confirm these settings have no bearing on experiencing the error (without them still happens):
```
        encrypt: true,
        trustServerCertificate: true,
        useColumnNames: true,
        enableArithAbort: true
```

**Comment by mShan0** [comment]
Our refactor of the NTLM logic introduced a bug to Tedious. We are working on a fix now. Thanks for reporting, will keep you updated.

**Comment by arthurschreiber** [comment]
@chriseaton This should have been fixed by #1446, and should be part of `tedious@14.6.1`. It's not tagged with `latest` yet, so you have to install it via `npm install tedious@14.6.1` or `npm install 14.6.1@next`. Please let us know if this fixes the issue you encountered.

Thanks for the bug report! 🙇‍♂️

**Comment by chriseaton** [comment]
Confirmed - resolved thanks!
